pkg/controller/elasticquota_test.go

```diff
@@ -0,0 +1,242 @@
+		min      v1.ResourceList
+		max      v1.ResourceList
+		nowUsed  v1.ResourceList
+		pods     []*v1.Pod
+		nextUsed v1.ResourceList
+	}{
+		{
+			name:    "no init containers pod",
+			eqName:  "eq1",
+			min:     makeResourceList().cpu(3).mem(5).obj(),
+			max:     makeResourceList().cpu(5).mem(15).gpu(1).obj(),
+			nowUsed: makeResourceList().obj(),
+			pods: []*v1.Pod{
+				makePod("pod1", v1.PodRunning).container(
+					makeResourceList().cpu(1).mem(2).gpu(1).obj()).obj(),
+				makePod("pod2", v1.PodPending).container(
+					makeResourceList().cpu(1).mem(2).gpu(0).obj()).obj(),
+			},
+			nextUsed: makeResourceList().cpu(1).mem(2).gpu(1).obj(),
+		},
+		{
+			name:    "have init containers pod",
+			eqName:  "eq2",
+			min:     makeResourceList().cpu(3).mem(5).obj(),
+			max:     makeResourceList().cpu(5).mem(15).obj(),
+			nowUsed: makeResourceList().cpu(1).mem(3).obj(),
+			pods: []*v1.Pod{
+				// cpu:2,mem: 4
+				makePod("pod1", v1.PodRunning).
+					container(
+						makeResourceList().cpu(1).mem(2).obj()).
+					container(
+						makeResourceList().cpu(1).mem(2).obj()).obj(),
+				// cpu: 3, men: 3
+				makePod("pod2", v1.PodRunning).
+					initContainerRequest(
+						makeResourceList().cpu(2).mem(1).obj()).
+					initContainerRequest(
+						makeResourceList().cpu(2).mem(3).obj()).
+					container(
+						makeResourceList().cpu(2).mem(1).obj()).
+					container(
+						makeResourceList().cpu(1).mem(1).obj()).obj(),
+			},
+			nextUsed: makeResourceList().cpu(5).mem(7).obj(),
+		},
+		{
+			name:    "update pods",
+			eqName:  "eq3",
+			min:     makeResourceList().cpu(3).mem(5).obj(),
+			max:     makeResourceList().cpu(5).mem(15).obj(),
+			nowUsed: makeResourceList().cpu(2).mem(4).gpu(1).obj(),
+			pods: []*v1.Pod{
+				// cpu:2,mem: 4
+				makePod("pod1", v1.PodRunning).
+					container(makeResourceList().cpu(1).mem(2).gpu(1).obj()).
+					container(makeResourceList().cpu(1).mem(2).obj()).obj(),
+				// cpu: 3, men: 3
+				makePod("pod1", v1.PodPending).
+					initContainerRequest(makeResourceList().cpu(2).mem(1).obj()).
+					initContainerRequest(makeResourceList().cpu(2).mem(3).obj()).
+					container(makeResourceList().cpu(2).mem(1).obj()).
+					container(makeResourceList().cpu(1).mem(1).obj()).obj(),
+			},
+			nextUsed: makeResourceList().cpu(0).mem(0).obj(),
+		},
+		{
+			name:     "min and max have the same fields",
+			eqName:   "eq4",
+			min:      makeResourceList().cpu(3).mem(5).obj(),
+			max:      makeResourceList().cpu(5).mem(15).obj(),
+			nowUsed:  makeResourceList().cpu(2).mem(4).obj(),
+			pods:     []*v1.Pod{},
+			nextUsed: makeResourceList().cpu(0).mem(0).obj(),
+		},
+		{
+			name:     "min and max have the different fields",
+			eqName:   "eq4",
+			min:      makeResourceList().cpu(3).mem(5).gpu(2).obj(),
+			max:      makeResourceList().cpu(5).mem(15).obj(),
+			nowUsed:  makeResourceList().cpu(2).mem(4).obj(),
+			pods:     []*v1.Pod{},
+			nextUsed: makeResourceList().cpu(0).mem(0).gpu(0).obj(),
+		},
```

Comment:
Added 2 tests to the description #[here](https://github.com/kubernetes-sigs/scheduler-plugins/pull/212#issuecomment-885447843)